### PR TITLE
Initiate switch from `build_script` and `test_script` to `script`

### PR DESCRIPTION
**Pull request by AyodeAwe**
The script used in wheel workflows is indicated by `script`, whereas the conda workflows use `build_script` and `test_script`.

This PR initiates a switch for the conda build and test workflows (cpp, python) to use the `script`  input name. After this PR is merged, PRs will be opened to all RAPIDS repositories which call these workflows to update the inputs from `build_script` and `test_script` (if these are present in the workflows) to `script`.

The rationale is that the naming convention of `build_script` and `test_script` appear rather redundant given the name of their respective workflows (`cpp_build`, `cpp_test`, etc.)

After these PRs are merged, another PR will be opened in this repository to remove the `build_script` and `test_script` references for these shared workflows.

This PR also adds a name for the conda build job, keeping it consistent with the named wheels build job.

### CODE REVIEW

File: .github/workflows/wheels-build.yaml [hunk 1]
```diff
@@ -18,6 +18,8 @@ on:
       script:
         required: true
```
Comment (raydouglass) [inline .github/workflows/wheels-build.yaml:19]: Should this input be switched to not required?

Comment (AyodeAwe) [reply .github/workflows/wheels-build.yaml:19]: Yeah looks like that should be removed.

File: .github/workflows/conda-cpp-build.yaml [hunk 2]
```diff
@@ -72,6 +74,7 @@ jobs:
 
           echo "MATRIX=${MATRIX}" | tee --append "${GITHUB_OUTPUT}"
   build:
+    name:  ${{ matrix.CUDA_VER }}, ${{ matrix.PY_VER }}, ${{ matrix.ARCH }}, ${{ matrix.LINUX_VER }}
```
Comment (ajschmidt8) [inline .github/workflows/conda-cpp-build.yaml:77]: was this change intended to be included in this PR? it's not mentioned in the PR body. I know others may have thoughts on what type of naming convention we use, so it might be best to postpone these changes for now.

Comment (AyodeAwe) [reply .github/workflows/conda-cpp-build.yaml:77]: should we at least have something there until there is a consensus on the naming convention we should have? I'm trying to clear out an inconsistency where the wheels build job is named but the conda job isn't.

I can include the change in the PR description

Comment (ajschmidt8) [reply .github/workflows/conda-cpp-build.yaml:77]: that's fine

**Review by ajschmidt8 — APPROVED** [review]
